modules/terraform/apply.go

```diff
@@ -60,3 +62,33 @@ func TgApplyAllE(t testing.TestingT, options *Options) (string, error) {
 
 	return RunTerraformCommandE(t, options, FormatArgs(options, "apply-all", "-input=false", "-lock=false", "-auto-approve")...)
 }
+
+// ApplyAndIdempotent runs terraform init and apply with the given options and return stdout/stderr from the apply command. It then runs
+// plan again and will fail the test if plan requires additional changes. Note that this method does NOT call destroy and assumes
+// the caller is responsible for cleaning up any resources created by running apply.
+func ApplyAndIdempotent(t testing.TestingT, options *Options) string {
+
+	out, err := ApplyAndIdempotentE(t, options)
+	require.NoError(t, err)
+
+	return out
+}
+
+// ApplyAndIdempotentE runs terraform init and apply with the given options and return stdout/stderr from the apply command. It then runs
+// plan again and will fail the test if plan requires additional changes. Note that this method does NOT call destroy and assumes
+// the caller is responsible for cleaning up any resources created by running apply.
+func ApplyAndIdempotentE(t testing.TestingT, options *Options) (string, error) {
+	out, err := InitAndApplyE(t, options)
+
+	if err != nil {
+		return "", err
+	}
+
+	exitCode := PlanExitCode(t, options)
```

Comment:
This should call `PlanExitCodeE` and return the error if it hits one.

Reply:
yip, makes sense.